src/NuGet.Core/NuGet.Common/Telemetry/INuGetTelemetryService.cs

```diff
@@ -1,13 +1,17 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-
-using System;
-
 namespace NuGet.Common
 {
+    /// <summary> Abstraction of NuGet telemetry service. </summary>
     public interface INuGetTelemetryService
     {
+        /// <summary> Send a <see cref="TelemetryEvent"/> to VS telemetry. </summary>
+        /// <param name="telemetryData"> Telemetry event to send. </param>
         void EmitTelemetryEvent(TelemetryEvent telemetryData);
```

Comment:
If you look at the references of the EmitTelemetry event you'll see that there's some telemetry events that are not emitting a telemetry marker. Some of those are gonna be particularly interesting for rich metadata, because they don't overlap.

Specifically look at https://github.com/NuGet/NuGet.Client/blob/721908c643a7806b630774e76b53e37507fec062/src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs#L257

Reply:
I see, those were not following `using var telemetry = TelemetryActivity.Create(...` pattern. I will convert them to proper activities, so we can mark them as well. I will also have PR to emit markers for every interval.

First I wrote this, but then I looked at code and came up with paragraph above 🙂:
> We can consume regular telemetry events in RPS. I am adding markers to be able to find _start_ of the TelemetryActivity in RPS. For events which are not representing timespan, we do not need markers.

Reply:
If you want you can do that in a future PR.
It'd allow for a faster turnaround.